examples/hardware_decoding/main.go

```diff
@@ -0,0 +1,195 @@
+	for _, is := range inputFormatContext.Streams() {
+		var err error
+
+		// Only process video
+		if is.CodecParameters().MediaType() != astiav.MediaTypeVideo {
+			continue
+		}
+
+		// Create stream
+		s := &stream{inputStream: is}
+
+		// Find decoder
+		if s.decCodec = astiav.FindDecoder(is.CodecParameters().CodecID()); s.decCodec == nil {
+			log.Fatal(errors.New("main: codec is nil"))
+		}
+
+		// Alloc codec context
+		if s.decCodecContext = astiav.AllocCodecContext(s.decCodec); s.decCodecContext == nil {
+			log.Fatal(errors.New("main: codec context is nil"))
+		}
+		defer s.decCodecContext.Free()
+
+		codec_hardware_configs, err := s.decCodec.HardwareConfigs(hardwareDeviceType)
+		if err != nil {
+			log.Fatal(fmt.Errorf("main: no codec hardware config found: %w", err))
+		}
+
+		for _, p := range codec_hardware_configs {
+			if p.MethodFlags().Has(astiav.CodecHardwareConfigMethodHwDeviceCtx) && p.HardwareDeviceType() == hardwareDeviceType {
+				hardwarePixelFormat = p.PixelFormat()
+				break
+			}
+		}
+
+		if hardwarePixelFormat == astiav.PixelFormatNone {
+			log.Fatal(fmt.Errorf("main: Decoder %s does not support device type", hardwareDeviceType.String()))
+		}
+
+		// Update codec context
+		if err := is.CodecParameters().ToCodecContext(s.decCodecContext); err != nil {
+			log.Fatal(fmt.Errorf("main: updating codec context failed: %w", err))
+		}
+
+		// Create hardware device context
+		hdc, err := astiav.CreateHardwareDeviceContext(hardwareDeviceType, "", nil)
+		if err != nil {
+			log.Fatal(fmt.Errorf("main: creating hardware device context failed: %w", err))
+		}
+		s.decCodecContext.SetHardwareDeviceContext(hdc)
+
+		// Open codec context
+		if err := s.decCodecContext.Open(s.decCodec, nil); err != nil {
+			log.Fatal(fmt.Errorf("main: opening codec context failed: %w", err))
+		}
+
+		// Add stream
+		streams[is.Index()] = s
+	}
+
+	// Loop through packets
+	for {
+		// Read frame
+		if err := inputFormatContext.ReadFrame(pkt); err != nil {
+			if errors.Is(err, astiav.ErrEof) {
+				break
+			}
+			log.Fatal(fmt.Errorf("main: reading frame failed: %w", err))
+		}
+
+		// Get stream
+		s, ok := streams[pkt.StreamIndex()]
+		if !ok {
+			continue
+		}
+
+		// Send packet
+		if err := s.decCodecContext.SendPacket(pkt); err != nil {
+			log.Fatal(fmt.Errorf("main: sending packet failed: %w", err))
+		}
+
+		// Loop
+		for {
+			// Receive frame
+			if err := s.decCodecContext.ReceiveFrame(hardwareFrame); err != nil {
+				if errors.Is(err, astiav.ErrEof) || errors.Is(err, astiav.ErrEagain) {
+					break
+				}
+				log.Fatal(fmt.Errorf("main: receiving frame failed: %w", err))
+			}
+
+			// Get frame
+			var finalFrame *astiav.Frame
+			if hardwareFrame.PixelFormat() == hardwarePixelFormat {
+				// Transfer hardware data
+				if err := hardwareFrame.TransferHardwareData(softwareFrame); err != nil {
+					log.Fatal(fmt.Errorf("main: Unable to transfer frame from gpu: %w", err))
```

Comment:
Could you update the error to `fmt.Errorf("main: transferring hardware data failed: %w", err)`